### PR TITLE
[ENHANCEMENT] postgreSQL Support

### PR DESCRIPTION
**Pull request by beaudurrant**
This PR includes additions to use existing migrations with postgreSQL:

- Added postgreSQL support
- Updated workflow to include postgreSQL matrix
- Updated Functions to use withVariable, withParameter, withDataUse & withDeterminism
- Updated Handler tests to use migration files from `/tests/Migrations` for postgreSQL & mySQL

### CODE REVIEW

File: tests/Migrations/20200602_create_user_counts_view.php [hunk 1]
```diff
@@ -1,4 +1,4 @@
 <?php
 
 return Exo\ViewMigration::create('user_counts')
-    ->withBody('select count(distinct id) as user_count from test.users');
+    ->withBody('SELECT COUNT(id) AS user_count FROM users');
```
Comment (Glenn-Ewing) [inline tests/Migrations/20200602_create_user_counts_view.php:4]: Did you mean to drop Distinct? 
It was not dropped in tests/Migrations/20200605_create_user_counts_view_with_context.php
Unless its needed for specific testing, we should keep these consistent.

**Review by Glenn-Ewing — COMMENTED** [review]
@beaudurrant Great work, thanks for this. I made one note on consistency, but otherwise, things look good from what I see. Let me know if the item I mentioned was intentional or not.

**Review by Glenn-Ewing — APPROVED** [review]
Looks good @beaudurrant , thank you!

**Comment by joshmcrae** [comment]
Solid work @beaudurrant! Thanks for this.

**Pull request merged**